crates/next-api/src/middleware.rs

```diff
@@ -123,7 +123,7 @@ impl MiddlewareEndpoint {
 
         let userland_module = self.userland_module();
 
-        let config = parse_config_from_source(userland_module);
+        let config = parse_config_from_source(userland_module).await?;
```

Comment:
For parallelism reasons it would be better to await config later.

You could add `let config = config.await?` in line 140

Reply:
Done!